### ISSUE TITLE
nginx-proxy network create in compose?

### ISSUE THREAD

**Issue by frostyandy2k** [ekkis/nginx-proxy-LE-docker-compose#1]
Hi, first of, this project is amazing, thank you for putting in the work!

I have a question, why can't we create the network in the docker-compose file for nginx and need to create it via the docker network create command?

```
docker network create --subnet=172.28.0.0/16 --opt com.docker.network.bridge.name=nginx-proxy nginx-proxy
```
Is there a problem with the naming or the way compose creates networks?

**Comment by ekkis** [comment]
the reason is primarily that you’ll want to share the same network across docker-compose files.  so you can keep your services in one docker-compose file and the nginx services in another one but they get to share networks

**Comment by frostyandy2k** [comment]
AFAIK that scenario is also possible by using the following snippet in the docker-compose where nginx is running
```
networks:
   nginx-proxy:
    driver: bridge
    ipam:
      driver: default
      config:
      -
        subnet: 172.28.0.0/16
```
Source: https://docs.docker.com/compose/compose-file/#ipv4address-ipv6address

**Comment by ekkis** [comment]
I like your idea as it would eliminate the extra step, but I can't make it work.  when I try it as you indicate I get an error:

```
$ docker-compose up -d
Creating network "nginxproxyledockercompose_nginx-proxy" with driver "bridge"
Creating nginx-proxy ... 
Creating nginx-proxy ... error

ERROR: for nginx-proxy  Cannot create container for service nginx-proxy: Conflict. The container name "/nginx-proxy" is already in use by container "578779072fe2dc295ad6569b737ceb3d66c8e57532a50d20a1039a69e4b9b604". You have to remove (or rename) that container to be able to reuse that name.

ERROR: for nginx-proxy  Cannot create container for service nginx-proxy: Conflict. The container name "/nginx-proxy" is already in use by container "578779072fe2dc295ad6569b737ceb3d66c8e57532a50d20a1039a69e4b9b604". You have to remove (or rename) that container to be able to reuse that name.
ERROR: Encountered errors while bringing up the project.
```

so it crashes the container creation but it does also create the network.  unfortunately it names it _nginxproxyledockercompose_nginx-proxy_ instead of just _nginx-proxy_.  I tried using the `name:` key in various places but it won't accept it and the docs are not very helpful.  it appears that key only works for external networks

thoughts?

**Comment by ekkis** [comment]
seems others have run into this issue: https://github.com/docker/compose/issues/3736

**Comment by frostyandy2k** [comment]
Seems everyone is using pre-existing networks in such a case.

The project_name+network_name_from_compose might be an issue because it varies depending on the folder name. Folder name change would require changing the compose files. I did forget about this.

I looked up removing the compose project name from the network or services, but it seems that is the way to go. I would not want to make workarounds just for this.

As for the Error:
`ERROR: for nginx-proxy  Cannot create container for service nginx-proxy: Conflict. The container name "/nginx-proxy" is already in use`

You just need to delete the existing containers, it seems compose doesn't like it when we fiddle with the network. It'd do a `docker-compose down` first and then an up.

I guess this issue can be closed until there is a way to create truly custom named networks with compose. I'm currently experimenting with Rancher, if there is a way to do it in rancher files, i'll post it here.
